src/test/java/chambre/morsecode/WordleControllerTest.java

```diff
@@ -2,28 +2,110 @@
 
 import org.junit.jupiter.api.Test;
 
+import javax.swing.*;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
 class WordleControllerTest {
     @Test
     public void addLetter() {
-        //asserts the letter has been added to the word
+        //given
+        WordleGame game = mock();
+        WordleDictionary dictionary = mock();
+        JLabel[][] labels = new JLabel[][]{{
+                mock(), mock(), mock(), mock(), mock(),
+        }, {
+                mock(), mock(), mock(), mock(), mock(),
+        }, {
+                mock(), mock(), mock(), mock(), mock(),
+        }, {
+                mock(), mock(), mock(), mock(), mock(),
+        }, {
+                mock(), mock(), mock(), mock(), mock(),
+        }, {
+                mock(), mock(), mock(), mock(), mock(),
+        },
+        };
+        JButton enter = mock();
+        JButton back = mock();
+        WordleController controller = new WordleController(game, dictionary, labels, enter, back);
+
+        //when
+        controller.addLetter("A");
 
-        //asserts the letter shows up in the right grid box on the screen
+        //then
+        verify(labels[0][0]).setText("A");
+        assertEquals("A", controller.theGuess.toString());
     }
 
     @Test
     public void enterGuess() {
+        //given
+        WordleGame game = mock();
+        WordleDictionary dictionary = mock();
+        JLabel[][] labels = new JLabel[][]{{
+                mock(), mock(), mock(), mock(), mock(),
+        }, {
+                mock(), mock(), mock(), mock(), mock(),
+        }, {
+                mock(), mock(), mock(), mock(), mock(),
+        }, {
+                mock(), mock(), mock(), mock(), mock(),
+        }, {
+                mock(), mock(), mock(), mock(), mock(),
+        }, {
+                mock(), mock(), mock(), mock(), mock(),
+        },
+        };
+        JButton enter = mock();
+        JButton back = mock();
+        WordleController controller = new WordleController(game, dictionary, labels, enter, back);
+
+        //when
+        for (int i = 0; i < game.getWordleWord().length() - 1; i++) {
+            controller.addLetter(game.getWordleWord().substring(i, i + 1));
+        }
```

Comment:
What is going on here?

Reply:
I am adding all the letters of the correct wordle word so that I can guarantee the word will be correct and then check that all the boxes are green.

Reply:
`game` is a mock.
`getWordleWord()` returns nothing.

Reply:
oh.... how can I use the mock to figure out what each color is supposed to be so that I can verify the correct color?

Reply:
Just call addLetter with a letter that you choose.